File: conversions/word-wrap/elide-word-wrap.rb
#!/usr/bin/env ruby

# Ruby Integration Example for elide-word-wrap
#
# This demonstrates calling the TypeScript word-wrap implementation from Ruby
# using Elide's polyglot capabilities.
#
# Benefits:
# - One word-wrap implementation shared across TypeScript and Ruby
# - Consistent wrap words to specified line width across microservices
# - Perfect for Rails, Sidekiq, background workers

puts "=== Ruby Calling TypeScript word-wrap ===\n"

# NOTE: Exact syntax depends on Elide's Ruby polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Ruby support is ready

# Assuming Elide provides something like:
# word_wrap_module = Elide.require('./elide-word-wrap.ts')

# Example: Basic usage
# result = word_wrap_module.default()
# puts "Result: #{result}"

puts "Real-world use case:"
puts "- Ruby worker uses wrap words to specified line width"
puts "- Uses same TypeScript implementation as Node.js API"
puts "- Guarantees consistent behavior across services"
puts "- Perfect for text formatting, terminal output, email composition"
puts ""

puts "Example: Microservices Architecture"
puts "┌─────────────────────────────────────┐"
puts "│   Elide word-wrap (TypeScript)         │"
puts "│   conversions/word-wrap/elide-word-wrap.ts│"
puts "└─────────────────────────────────────┘"
puts "         ↓           ↓           ↓"
puts "    ┌────────┐  ┌────────┐  ┌────────┐"
puts "    │ Node.js│  │  Ruby  │  │ Rails  │"
puts "    │  API   │  │ Worker │  │  App   │"
puts "    └────────┘  └────────┘  └────────┘"
puts "         ↓           ↓           ↓"
puts "    Same wrap words to specified line width everywhere!"
puts ""

puts "Problem Solved:"
puts "Before: Ruby custom + JavaScript = different behaviors"
puts "After: One Elide implementation = 100% consistent"
puts ""

puts "Rails Integration (when ready):"
puts "  # config/initializers/word_wrap.rb"
puts "  WORD_WRAP = Elide.require('./elide-word-wrap.ts')"

File: converted/utilities/word-wrap/elide-word-wrap.py
#!/usr/bin/env python3
"""
Python Integration Example for elide-word-wrap

This demonstrates calling the TypeScript word-wrap implementation from Python
using Elide's polyglot capabilities.

Benefits:
- One word-wrap implementation shared across TypeScript and Python
- Consistent wrap words to specified line width across services
- No Python custom implementation needed
- Perfect for text formatting, terminal output, email composition
"""

# NOTE: Exact syntax depends on Elide's Python polyglot API (currently alpha)
# This is a CONCEPTUAL example - adjust when Python support is ready

# Assuming Elide provides something like:
# from elide import require
# word_wrap_module = require('./elide-word-wrap.ts')

print("=== Python Calling TypeScript word-wrap ===\n")

# Example: Basic usage
# result = word_wrap_module.default()
# print(f"Result: {result}")

print("Real-world use case:")
print("- Python service uses wrap words to specified line width")
print("- Uses same TypeScript implementation as Node.js API")
print("- Guarantees consistent behavior across entire stack")
print("- Perfect for text formatting, terminal output, email composition")
print()

print("Example: Microservices Architecture")
print("┌─────────────────────────────────────┐")
print("│   Elide word-wrap (TypeScript)         │")
print("│   conversions/word-wrap/elide-word-wrap.ts│")
print("└─────────────────────────────────────┘")
print("         ↓                   ↓")
print("    ┌────────┐          ┌────────┐")
print("    │ Node.js│          │ Python │")
print("    │  API   │          │Pipeline│")
print("    └────────┘          └────────┘")
print("         ↓                   ↓")
print("    Same wrap words to specified line width everywhere!")
print()

print("Problem Solved:")
print("Before: Python custom implementation + JavaScript = inconsistencies")
print("After: One Elide implementation = 100% consistent")
print()

print("When Elide Python API is ready, usage will be:")
print("  from elide import require")
print("  word_wrap = require('./elide-word-wrap.ts')")
print("  # Use it directly!")
